### ISSUE TITLE
Still frustrated with syncing

### ISSUE THREAD

**Issue by pentool** [HappenApps/Quiver#425]
I have posted a few weeks back about syncing issues (https://github.com/HappenApps/Quiver/issues/287). Currently I'm in frustration again.

I just did a clean install on my desktop mac on January 30th. It seems my syncing somehow stalled between the desktop and my laptop (using iCloud). On the fresh system, the Quiver library is dated February 2. On my laptop it's January 28.

After I did a clean install, I went to Preferences > Sync > Open Another Library and selected the library on iCloud. It loaded fine. Since the 30th I've added a couple of changes and new items in Quiver on the desktop mac, but still my laptop is lagging behind by 5 days.

On the MacBook (which has the older stuff) I tried selecting the library on iCloud again (in the Preferences), but it still not syncing up. I tried modifying entries on both machines hoping that somehow the sync will kick in by sensing the changes. It's been near 40 minutes I'm playing with this but the sync is not happening.

I do like Quiver a lot, but this syncing issue is just really frustrating. In my original post (above) someone pointed out to use Dropbox. Right. But I don't want to install yet another service for just a single use. All my stuff is syncing with iCloud, I see no point switching to Dropbox just for this. I really wish Quiver would sync with iCloud the way all other apps do (where I can go to the System Preferences > iCloud and I can select the apps what to sync) as opposed to the drive itself. This seems highly unreliable for some reason (...or I am just simply giving off really bad vibes and make Quiver fall into depression)

Quiver 3.0 (66)

Mac mini late 2012
El Crapitan 10.11.3

MacBook Pro early 2015
Yo Semite 10.10.5

**Comment by ylian** [comment]
I honestly have no idea why iCloud Drive doesn't sync properly. It should, and that's what iCloud Drive is designed to do.

When I implemented syncing in Quiver 2, I made the design decision to store data as files, and rely on the cloud service to do the actual syncing. This is a simpler design and can support multiple cloud services with one implementation. Supporting iCloud (not iCloud Drive) like other apps would require a parallel implementation of all the data models and syncing logic, and that would make the implementations much more complicated.

I try to keep things simple, so I can keep building Quiver without being overwhelmed by it. I am certainly not the best Mac developer out there (I learned all these in the last two years), so I apologize for not knowing how to fix this at the moment.

**Comment by drumanagh** [comment]
Unfortunately, iCloud is just plain bad at handling file syncing. Unlike Dropbox or basically every other sync service out there, there are constant unexplained delays in its syncing (and you can find plenty of complaints about this on the web).

**Comment by paulelms** [comment]
iCloud works just fine for me. I suggest you to reboot the system.

**Comment by pentool** [comment]
It started working again. But yeah, it took a while. In the evening I just gave up but when I tried it in the morning it synced up pretty fast.

> I am certainly not the best Mac developer out there (I learned all these in the last two years), so I apologize for not knowing how to fix this at the moment.

There's no need to apologize. I was always impressed with what a great job you did/do with responding, fixing bugs, and adding new features. And all this just under 2 years, that's impressive!

I was heading out Saturday with my laptop and wanted to make sure it's synced. So Friday evening I was doing it and it just didn't want to cooperate, so I wasn't in a happy place. But eventually it succeeded in the morning.
